### PR TITLE
parse cloud formation templates

### PR DESCRIPTION
**Pull request by sjmiller609**
##### SUMMARY
Include a new lookup function for parsing cloud formation templates

I closed PR 44123 because I was not getting a response.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
lookup cloudformation

##### ANSIBLE VERSION
2.6.1

### CODE REVIEW

File: lib/ansible/plugins/lookup/cloudformation_parse.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    description:
+      - The parsed cloudformation file(s)
+    type: list
+"""
+
+# This module allows us to parse cfn including special tags
+# The module is supported by awslabs
+# see 'load' function here
+# https://github.com/awslabs/aws-cfn-template-flip/blob/master/cfn_flip/__init__.py
+from cfn_flip import load as cfn_load
```
Comment (gundalow) [inline lib/ansible/plugins/lookup/cloudformation_parse.py]: Please add a try/expect around this import like in https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/lookup/consul_kv.py#L108-L123

File: lib/ansible/plugins/lookup/cloudformation_parse.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# (c) 2018, Steven Miller
+# (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
```
Comment (gundalow) [inline lib/ansible/plugins/lookup/cloudformation_parse.py:7]: Please add `requirements:` for `cfn_flip`

Example: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/lookup/consul_kv.py#L20-L21

File: lib/ansible/plugins/lookup/cloudformation_parse.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    lookup: cloudformation_parse
+    author: Steven Miller (@sjmiller609)
+    version_added: "2.6"
```
Comment (gundalow) [inline lib/ansible/plugins/lookup/cloudformation_parse.py]: 2.8

File: lib/ansible/plugins/lookup/cloudformation_parse.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    short_description: Parse cloudformation files
```
Comment (gundalow) [inline lib/ansible/plugins/lookup/cloudformation_parse.py]: CloudFormation

Applies to multiple places

**Review by gundalow — CHANGES_REQUESTED** [review]
Few minor comments, can you also please add a `changelog/fragment` file

**Comment by sjmiller609** [comment]
@gundalow Thank you for your review. I have taken care of your comments.

**Comment by sjmiller609** [comment]
@gundalow This PR is ready for final review

**Pull request closed without merging**